third_party/move/move-compiler-v2/tests/reference-safety/bug_12394_missing_freeze.move

```diff
@@ -0,0 +1,14 @@
+module 0x815::m {
+    fun t2(u1: &mut u64, u2: &mut u64): (&mut u64, &mut u64) {
+        (u1, u2)
+    }
+    fun test_4() {
+        let _x: &u64;
+        let _y: &u64;
+        (_x, _y) = t2(&mut 3, &mut 4);
+    }
+
+    fun test_5() {
+        let (_x, _y): (&u64, &u64) = t2(&mut 3, &mut 4);
+    }
+}
```

Comment:
Could we please add the following (transactional) test case:

```
//# publish
module 0xc0ffee::m {
    public fun test(): u64 {
        let a = 1;
        let (x, y): (&u64, &u64) = (&mut a, freeze(&mut a));
        *x + *y
    }
}

//# run 0xc0ffee::m::test
```

On this branch, the above code gets a complier error in v1, but successfully executes in v2.

If you think this should be a bug filed, let me know and I can do so.

Reply:
Added it, and that is what I'm seeing here too. Seems to me a bug in v1 borrow analysis, which wee aren't going to fix. I added a comment to clarify this in the test case.